pkg/lifecycle/service.go

```diff
@@ -431,6 +431,8 @@ func (s *Service) buildRunnablePipeline(
 ) (*runnablePipeline, error) {
 	nodes, err := s.buildNodes(ctx, pl)
 	if err != nil {
+		delete(s.pipelines.List(ctx), pl.ID)
```

Comment:
Is there a special reason for not using s.pipelines.Delete()?

Reply:
newer version of conduit handles incorrectly provisioned pipelines and makes this line redundant so I have removed it 👍